### ISSUE TITLE
SwaggerEditor@next: autocompletion works incorectly

### ISSUE THREAD

**Issue by char0n** [swagger-api/swagger-editor#4216]
Autocompletion works inproperly on next editor. If I write e.g. aaa it shows all words which contains few a. I expect it should show only words which contains ‘aaa' in a row

**Steps To Reproduce**

1. Open next editor (OAS 3 or OAS 3.1)
2. Start writing e.g. 'aaa'

**Expected results**
I expect it should show only words which contains ‘aaa' in a row. - see screen3 and screen4(examples from SH ACE editor)

**Actual results**
it shows all words which contains few a - see screen1 and screen2

**screen1**
![screen1](https://github.com/swagger-api/swagger-editor/assets/193286/8da66511-c553-4d18-8a8c-0813dc48ed1c)

**screen2**
![screen2](https://github.com/swagger-api/swagger-editor/assets/193286/dbf58407-bc15-424a-825b-9e437acd6796)

**screen3**

![screen3](https://github.com/swagger-api/swagger-editor/assets/193286/1b6e6512-16fe-4cd7-89fd-45a952f16624)

**screen4**

![screen4](https://github.com/swagger-api/swagger-editor/assets/193286/d26050c5-65e4-498e-b00b-1f5676c18d71)

**Comment by frantuma** [comment]
@kowalczyk-krzysztof @char0n @ponelat 

[This APIDom PR #2954](https://github.com/swagger-api/apidom/pull/2954) introduces support for language service "strict" filtering, by setting `enableLSPFilter: true` in `completionContext` (within `languageServiceContext`):

```typescript

export class ApiDOMWorker {
  static apiDOMContext = {
    validatorProviders: [],
    completionProviders: [],
    performanceLogs: false,
    logLevel: apidomLS.LogLevel.WARN,
    defaultLanguageContent: {
      namespace: 'asyncapi',
    },
    completionContext: {
      maxNumberOfItems: 100,
      enableLSPFilter: true,
    },
  };

```
An mentioned in https://github.com/swagger-api/apidom/pull/2954#issue-1806082963 and copying from comment:

> This is an opinionated behaviour, as the default for LSP completions is to leave the filtering logic to the editor/client; Monaco editor always performs fuzzy matching against the filterText value and/or the current word, and in the https://github.com/swagger-api/swagger-editor/issues/4216 it still displays non "strict matching" items, which is possibly a desired behavior in many cases (e.g it still provides hints in case of typos etc).

> This is [quite debated though](https://github.com/microsoft/language-server-protocol/issues/648).

As a user I would probably favor the standard behavior with fuzzy matching, and IMO this would be kept disabled by default, still allowing apps to enable it, maybe even providing a user option. This is not a strong opinion though.

https://github.com/swagger-api/swagger-editor/pull/4289 adds a comment/placeholder demonstrating usage, in case decision is to implement it by default we will update PR to enable the flag.

**Comment by frantuma** [comment]
For reference see also https://github.com/microsoft/vscode/issues/141224 and https://github.com/microsoft/vscode/issues/64367

**Comment by char0n** [comment]
> As a user I would probably favor the standard behavior with fuzzy matching, and IMO this would be kept disabled by default, still allowing apps to enable it, maybe even providing a user option. This is not a strong opinion though.

Yep, I do agree with keeping the current behavior (fuzzy matching) and giving the `editor-monaco-language-apidom` plugin capability to configure it on mount. I'll follow up with https://github.com/swagger-api/swagger-editor/pull/4289 and allow to make this configurable, with default value of `false`.

**Comment by frantuma** [comment]
closing as completed in https://github.com/swagger-api/apidom/pull/2954 and #4290
